src/constants.js

```diff
@@ -1,11 +1,40 @@
 const status = {
-  sqliteOk:      0,
-  sqliteRow:     100,
-  sqliteDone:    101,
-  stmtLimit:     1000,
-  noStmt:        1001,
-  databaseLimit: 1002,
-  noDatabase:    1003,
+  sqliteOk:          0,   // Successful result
+  sqliteError:       1,   // Generic error
+  sqliteInternal:    2,   // Internal logic error in SQLite
+  sqlitePerm:        3,   // Access permission denied
+  sqliteAbort:       4,   // Callback routine requested an abort
+  sqliteBusy:        5,   // The database file is locked
+  sqliteLocked:      6,   // A table in the database is locked
+  sqliteNoMem:       7,   // A malloc() failed
+  sqliteReadOnly:    8,   // Attempt to write a readonly database
+  sqliteInterrupt:   9,   // Operation terminated by sqlite3_interrupt()
+  sqliteIOErr:       10,  // Some kind of disk I/O error occurred
+  sqliteCorrupt:     11,  // The database disk image is malformed
+  sqliteNotFound:    12,  // Unknown opcode in sqlite3_file_control()
+  sqliteFull:        13,  // Insertion failed because database is full
+  sqliteCantOpen:    14,  // Unable to open the database file
+  sqliteProtocol:    15,  // Database lock protocol error
+  sqliteEmpty:       16,  // Internal use only
+  sqliteSchema:      17,  // The database schema changed
+  sqliteTooBig:      18,  // String or BLOB exceeds size limit
+  sqliteConstraint:  19,  // Abort due to constraint violation
+  sqliteMismatch:    20,  // Data type mismatch
+  sqliteMisuse:      21,  // Library used incorrectly
+  sqliteNoLFS:       22,  // Uses OS features not supported on host
+  sqliteAuth:        23,  // Authorization denied
+  sqlietFormat:      24,  // Not used
+  sqliteRange:       25,  // 2nd parameter to sqlite3_bind out of range
+  sqliteNotADB:      26,  // File opened that is not a database file
+  sqliteNotice:      27,  // Notifications from sqlite3_log()
+  sqliteWarning:     28,  // Warnings from sqlite3_log()
+  sqliteRow:         100, // sqlite3_step() has another row ready
+  sqliteDone:        101, // sqlite3_step() has finished executing
+
+  stmtLimit:         1000,
```

Comment:
Sure thing, make sense.
To be honest, I don't know what they stand for, I would appreciate help here. :stuck_out_tongue:

Reply:
The interface is designed to never leak any memory (with the sole exception of using the exposed `malloc`). To achieve this, open database handles and open transactions are stored in fixed sized registries (hence they all have id's instead of pointers).

`stmtLimit` -> statement limit was reached: the statement registry is full, no more statements can be opened
`noStms` -> registry entry at this id is empty
`databaseLimit` -> database limit was reached: the database registry is full, no more databases can be opened
`noDatabase` -> registry entry at this id is empty

Reply:
Thanks for the explanation.

Added comments.